api/envoy/api/v2/cds.proto

```diff
@@ -350,6 +351,18 @@ message Cluster {
     // to further constrain resource use. See also
     // :ref:`minimum_ring_size<envoy_api_field_Cluster.RingHashLbConfig.minimum_ring_size>`.
     google.protobuf.UInt64Value maximum_ring_size = 4 [(validate.rules).uint64 = {lte: 8388608}];
+
+    // When true, the host's logical name is used to place it in the hash ring.
+    // Otherwise it will use the resolved address.
+    bool use_logical_name = 5;
```

Comment:
Logical name is not an API concept so far. Is the intention that we are using the DNS name for hosts that are resolved this way? If so, how does this apply to EDS and other LBs?